### DISCUSSION TITLE
Enhancement: Option add DALL-E and any Plugin to the list of Providers, alongside OpenAI, Gemini,..., Plugins

### DISCUSSION THREAD

**Discussion by raymelon** [danny-avila/LibreChat#3532] [Feature Requests & Suggestions]
### Enhancement: Option add DALL-E and any Plugin to the list of Providers. Provider dropdown will look like this: OpenAI, Gemini, DALL-E, Plugins

Hi, first of all great job in building LibreChat. I am using it and so far, I am very impressed.

I wonder if it is possible to list DALL-E in provider dropdown, alongside OpenAI, Gemini, ..., Plugins.

I know that DALL-E is a plugin and it takes three steps to install and activate a plugin. I am hoping that by switching an environment variable, these steps will be skipped and make the plugin available for every LibreChat user.

I also know that every plugin has a preset, so ideally, making DALL-E available in the provider dropdown is possible in some way.

The behavior of enabling an environment variable will be:

1. Installing DALL-E plugin automatically for every LibreChat user
2. Making a DALL-E preset automatically
3. Adding "DALL-E" option on Provider dropdown
4. Enabling the preset once "DALL-E" option is clicked

But it can be more than that. Let me know.

That's all.
Cheers everyone!

### More details

Please see image:

![image](https://github.com/user-attachments/assets/91fe7227-cf51-4f73-8213-4a683f9825c3)

### Which components are impacted by your request?

UI, Plugins

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by danny-avila** [comment]
Thanks! So one of the main ways to do this is through a feature called model specs. This also gets into the territory of media/tool endpoints which is an idea I’ve been kicking around for a while.

here’s the page on this feature: https://www.librechat.ai/docs/configuration/librechat_yaml/object_structure/model_specs#overview

**Reply by raymelon** [reply]
Hey @danny-avila! Appreciate the prompt response!

Skimmed a bit, and it looks like it will allow me to add a model.

Possible scenario for this is:

1. Modify `librechat.yaml`
2. Add a new model through `modelSpecs`
3. See the new model under models dropdown

If I will use `openAI` as `endpoint`, I will only see the `DALL-E` selection after I select `Open AI` as provider right?
